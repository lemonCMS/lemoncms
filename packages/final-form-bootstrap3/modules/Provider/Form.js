import React from "react";
import PropTypes from "prop-types";
import { Form as FinalForm } from "react-final-form";
import arrayMutators from "final-form-arrays";
import ContextWrapper from "../Context/ContextWrapper";

const Form = props => {
  const { children, className, debug, readOnly, listen, ...formProps } = props;

  return (
    <FinalForm
      {...formProps}
      mutators={{
        ...arrayMutators
      }}
      render={({ handleSubmit, ...rest }) => {
        return (
          <ContextWrapper
            {...rest}
            debug={debug}
            listen={listen}
            readOnly={readOnly}
          >
            <form onSubmit={handleSubmit} className={className}>
              {children}
            </form>
          </ContextWrapper>
        );
      }}
    />
  );
};

Form.propTypes = {
  className: PropTypes.string,
  validate: PropTypes.func,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  debug: PropTypes.bool,
  readOnly: PropTypes.bool,
  listen: PropTypes.func,
  initialValues: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
      PropTypes.array,
      PropTypes.object,
      PropTypes.bool
    ])
  )
};

Form.defaultProps = {
  className: "",
  validate: () => {},
  listen: () => {},
  debug: false,
  readOnly: false,
  initialValues: {}
};

export default Form;
