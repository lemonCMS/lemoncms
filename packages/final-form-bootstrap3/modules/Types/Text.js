import React from "react";
import PropTypes from "prop-types";
import { FormControl } from "react-bootstrap";
import context from "../decorators/context";
import fieldGroup from "./fieldGroup";

const Text = props => {
  const { input, type, isDisabled } = props;
  return <FormControl {...input} type={type} disabled={isDisabled} />;
};

Text.propTypes = {
  input: PropTypes.shape({
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onBlur: PropTypes.func.isRequired,
    onFocus: PropTypes.func.isRequired,
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.bool,
      PropTypes.number
    ])
  }),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired,
  type: PropTypes.oneOf(["text", "email", "date", "datetime-local", "checkbox"])
};

Text.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null,
  type: "text"
};

export default context()(fieldGroup(Text));
