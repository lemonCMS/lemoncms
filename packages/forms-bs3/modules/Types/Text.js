import React from "react";
import PropTypes from "prop-types";
import { FormControl } from "react-bootstrap-v3";
import context from "../decorators/context";
import fieldGroup from "./fieldGroup";

const Text = props => {
  const { input, type } = props;
  return <FormControl {...input} type={type} />;
};

Text.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired,
  type: PropTypes.oneOf(["text", "email", "date", "datetime-local", "checkbox"])
};

Text.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null,
  type: "text"
};

export default context(fieldGroup(Text));
