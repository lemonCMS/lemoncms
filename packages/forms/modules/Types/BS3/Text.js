import React from "react";
import PropTypes from "prop-types";
import {
  FormGroup,
  ControlLabel,
  FormControl,
  HelpBlock
} from "react-bootstrap-v3";
import context from "../../decorators/context";
import field from "../../decorators/field";

const Text = props => {
  const { input, placeholder, computedInvalid, label, help } = props;

  return (
    <FormGroup validationState={computedInvalid ? "error" : null}>
      {label && <ControlLabel>{label}</ControlLabel>}
      <FormControl type="text" placeholder={placeholder} {...input} />
      <FormControl.Feedback />
      {help && <HelpBlock>{help}</HelpBlock>}
    </FormGroup>
  );
};

Text.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Text.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null
};

export default context(field(Text));
