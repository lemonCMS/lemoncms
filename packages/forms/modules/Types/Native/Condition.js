import React from 'react';
import PropTypes from 'prop-types';
import AppContext from '../../Context/AppContext';

const Condition = props => {
  const {
    context, children, hidden, visible
  } = props;
  if (hidden && context.checkCondition(hidden)) {
    return null;
  }
  if (visible && !context.checkCondition(visible)) {
    return null;
  }
  return children;
};

Condition.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
  context: PropTypes.shape({
    checkCondition: PropTypes.func
  }).isRequired,
  hidden: PropTypes.func,
  visible: PropTypes.func
};

Condition.defaultProps = {
  hidden: null,
  visible: null
};

export default props => (
  <AppContext.Consumer>{context => <Condition context={context} {...props} />}</AppContext.Consumer>
);
