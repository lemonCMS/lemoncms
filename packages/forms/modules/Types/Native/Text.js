import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import context from '../../decorators/context';
import field from '../../decorators/field';

const Text = props => {
  const {
    input, placeholder, formControl, computedInvalid
  } = props;
  return (
    <input
      type="text"
      placeholder={placeholder}
      className={classNames('ffc-input', { 'is-invalid': computedInvalid }, formControl)}
      {...input}
    />
  );
};

Text.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  formControl: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Text.defaultProps = {
  input: {},
  placeholder: null,
  formControl: null
};

export default context(field(Text));
