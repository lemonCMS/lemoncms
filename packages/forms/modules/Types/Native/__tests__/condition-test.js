import React from 'react';
import renderer from 'react-test-renderer';
import Form from '../../../Provider/Form';
import Text from '../Text';
import Condition from '../Condition';

const classes = {
  formGroup: 'form-group row',
  formLabel: 'form-label col-sm-5 col',
  formControl: 'form-control col-sm-7 col',
  formError: 'small offset-sm-5',
  formText: 'small offset-sm-5 text-muted'
};

beforeEach(() => {
  // Clear state
});

describe('Condition', () => {

  it('should render with children `visible`', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {}}
        >
          <Condition visible={() => true}>
            <Text
              name={"field2"}
              placeholder={"field2 placeholder"}
              {...classes}
            />
          </Condition>
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render without children `visible`', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {}}
        >
          <Condition visible={() => false}>
            <Text
              name={"field2"}
              placeholder={"field2 placeholder"}
              {...classes}
            />
          </Condition>
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render with children `hidden`', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {}}
        >
          <Condition hidden={() => false}>
            <Text
              name={"field2"}
              placeholder={"field2 placeholder"}
              {...classes}
            />
          </Condition>
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should render without children `hidden`', () => {
    const tree = renderer
      .create(
        <Form
          onSubmit={() => {}}
        >
          <Condition hidden={() => true}>
            <Text
              name={"field2"}
              placeholder={"field2 placeholder"}
              {...classes}
            />
          </Condition>
        </Form>);
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });
});
