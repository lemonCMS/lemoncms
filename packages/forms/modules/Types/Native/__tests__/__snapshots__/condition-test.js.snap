// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Condition should render with children \`hidden\` 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <input
      className="ffc-input form-control col-sm-7 col"
      name="field2"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="field2 placeholder"
      type="text"
      value=""
    />
  </div>
</form>
`;

exports[`Condition should render with children \`visible\` 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="ffc-field form-group row"
  >
    <input
      className="ffc-input form-control col-sm-7 col"
      name="field2"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="field2 placeholder"
      type="text"
      value=""
    />
  </div>
</form>
`;

exports[`Condition should render without children \`hidden\` 1`] = `
<form
  className=""
  onSubmit={[Function]}
/>
`;

exports[`Condition should render without children \`visible\` 1`] = `
<form
  className=""
  onSubmit={[Function]}
/>
`;
